src/main/scala/com/gu/newsletterlistcleanse/braze/BrazeClient.scala

```diff
@@ -1,55 +1,102 @@
 package com.gu.newsletterlistcleanse.braze
 
+import io.circe.Decoder
 import org.slf4j.{Logger, LoggerFactory}
-import scalaj.http._
+import sttp.client._
+import sttp.client.asynchttpclient.future.AsyncHttpClientFutureBackend
+
+import scala.concurrent.duration._
 import io.circe.parser.decode
 import io.circe.syntax._
 
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
 object BrazeClient {
 
-  private val timeout = 5000
+  private val timeout = 5000.seconds
+  implicit val sttpBackend = AsyncHttpClientFutureBackend(
+    options=SttpBackendOptions.connectionTimeout(timeout)
+  )
 
   val logger: Logger = LoggerFactory.getLogger(this.getClass)
   val brazeEndpoint = "https://rest.fra-01.braze.eu"
 
-  private def withClientLogging[A](info: => String)(block: => Either[BrazeError,A]): Either[BrazeError, A] = {
-    val result = block
+  private def withClientLogging[A](info: => String)(block: => Future[Either[BrazeError,A]]): Future[Either[BrazeError, A]] = {
+    val results = block
 
-    result.foreach { successResult =>
-      logger.info(s"BrazeClient success $info $successResult")
-    }
+    for {
+      result <- results
+      } {
+        result.foreach { successResult =>
+          logger.info(s"BrazeClient success: $info $successResult")
+        }
 
-    result.left.foreach { errorResult =>
-      logger.error(s"BrazeClient failure $info $errorResult")
-    }
+        result.left.foreach { errorResult =>
+          logger.error(s"BrazeClient failure: $info $errorResult")
+        }
+      }
```

Comment:
I think this might be slightly easier to read with

```scala
// calling foreach only executes the code once the future resolves
results.foreach {
  case Right(successResult) => logger.info(s"BrazeClient success: $info $successResult")
  case Left(errorResult) => logger.error(s"BrazeClient failure: $info $errorResult")
}
```

Reply:
Done!